UnitTests/ExternalClientTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using NUnit.Framework;
+using NamedPipeWrapper;
+using log4net.Appender;
+using log4net.Config;
+using log4net.Layout;
+using System.IO.Pipes;
+using System.IO;
+
+namespace UnitTests
+{
+    [TestFixture]
+    class ExternalClientTests
+    {
+        private static readonly log4net.ILog Logger =
+            log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+        static ExternalClientTests()
+        {
+            var layout = new PatternLayout("%-6timestamp %-5level - %message%newline");
+            var appender = new ConsoleAppender { Layout = layout };
+            layout.ActivateOptions();
+            appender.ActivateOptions();
+            BasicConfigurator.Configure(appender);
+        }
+
+        private const string PipeName = "external_client_test_pipe";
+
+        private NamedPipeServer<byte[]> _server;
+        private NamedPipeClientStream _client;
+
+        private DateTime _startTime;
+
+        private readonly ManualResetEvent _barrier = new ManualResetEvent(false);
+
+        #region Setup and teardown
+
+        [SetUp]
+        public void SetUp()
+        {
+            Logger.Debug("Setting up test...");
+
+            _barrier.Reset();
+
+            _server = new NamedPipeServer<byte[]>(PipeName, true);
+            _client = new NamedPipeClientStream("external_client_test_pipe");
+
+            _server.Error += ServerOnError;
+
+            _server.StartServerOnly();
```

Comment:
What's difference between calling _`server.StartServerOnly();` and `_server.Start();`?
On my environment, when I call `_server.Start()` I get the following exception:
```
Named pipe is broken or disconnected: System.UnauthorizedAccessException: Access to the path is denied.
   at System.IO.Pipes.NamedPipeServerStream.Create(String pipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode, PipeOptions options, Int32 inBufferSize, Int32 outBufferSize, PipeSecurity pipeSecurity, HandleInheritability inheritability, PipeAccessRights additionalAccessRights)
```